src/System.CommandLine/src/System/CommandLine/ArgumentSyntax.cs

```diff
@@ -52,12 +53,7 @@ private void Validate()
                 var helpText = GetHelpText();
                 Console.Error.Write(helpText);
 
-                // TODO: This should use Environment.Exit(0) but this API isn't available yet.
```

Comment:
You're completely right, this modification is a breaking change for external behavior of Parse method. However the library is experimental and hasn't been shipped with .NET framework or via NuGet. Thus the main idea was to change this quite unexpected external behavior of the method during experimental stage.

By the way, all mentioned in [README](https://github.com/dotnet/corefxlab/tree/master/src/System.CommandLine#why-a-new-library) command line parser libraries don't attempt to terminate application during parsing for any reason.